### PR TITLE
Add Gitpod configuration

### PR DESCRIPTION
**Pull request by AntonyMilneQB**
As in https://github.com/kedro-org/kedro/pull/1232, this PR adds Gitpod configuration files that makes it much easier to develop on this repo using Gitpod. I will update the contributing guide in a separate PR. Here's what you get on the kedro-viz Gitpod experience:

- [x] all JavaScript dependencies installed using latest 14.x Node.js 🎉 
- [x] frontend dev server running ✨ 
- [x] all Python dependencies installed 🐍 
- [x] backend dev server running on `demo-project` 🧪 
- [x] the checked out Kedro-Viz Python package installed in editable/develop mode 🧑‍💻
- [x] commits automatically DCO signed-off 👮

if you're using the in-browser VS Code, then the [ESlint extension](https://marketplace.visualstudio.com/items?itemName=dbaeumer.vscode-eslint) is also installed. We can easily add more extensions if there's others that people like to use.

## Tips
* You can launch Gitpod anywhere in our repo by prefixing the GitHub URL with `gitpod.io/#`. The [browser bookmarklet](https://www.gitpod.io/docs/browser-bookmarklet) or [browser extension](https://www.gitpod.io/docs/browser-extension) make it even easier
* If you're using the in-browser editor then [alter your settings](https://www.gitpod.io/docs/configure/browser-settings) to enable copy & paste and allow popups

## Checklist

- [x] Read the [contributing](/CONTRIBUTING.md) guidelines
- [ ] Opened this PR as a 'Draft Pull Request' if it is work-in-progress
- [ ] Updated the documentation to reflect the code changes
- [ ] Added new entries to the `RELEASE.md` file
- [ ] Added tests to cover my changes

<a href="https://gitpod.io/#https://github.com/kedro-org/kedro-viz/pull/735"></a>

### CODE REVIEW

File: .gitpod.yml [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+      gp sync-done build_complete
+      DANGEROUSLY_DISABLE_HOST_CHECK=true npm run start
+  - name: backend
+    before: echo PIP_USER=no >> ~/.bashrc && export PIP_USER=no
+    init: |
+      pip install -r package/test_requirements.txt -r demo-project/src/docker_requirements.txt --user
+      pip install https://github.com/kedro-org/kedro/archive/main.zip --no-deps --user --force-reinstall
+    command: | 
+      gp sync-await build_complete
+      pip install package --no-deps
```
Comment (AntonyMilneQB) [inline .gitpod.yml:21]: TODO: this should be `-e` really.

Comment (AntonyMilneQB) [reply .gitpod.yml]: Resolved by reverting to legacy Docker image.

**Comment by AntonyMilneQB** [comment]
As with the kedro PR, I've requested lots of reviews just for visibility. Don't worry about it if you're not intested!

**Review by tynandebold — APPROVED** [review]
Amazing! Let's get it in.

**Comment by studioswong** [comment]
thanks for setting this up @AntonyMilneQB  - love your fun emoji filled description! 👍 

Just a few questions to fulfill my curiosity:

1. I wonder how does gitpod know which node version to use when doing the npm install? (it seems magical as our npm build will fail on newer node versions 😆 )
2. What are the next steps we need to do for the kedro-viz github repo settings in order to enable the gitpod setup? ( i.e do we need to link this repo to a gitpod account so it will trigger deployment every time a branch / commit gets pushed?)
2. Are there anything that needs to be added to our CI/CD in order to trigger this gitpod deployment? ( or does it work similar to cloudflare / Vercel where on linking to the gitpod account, it will just 'listen 'to this repo and deploy automatically?) 

Thanks for setting this up and making this happen 😄

**Comment by AntonyMilneQB** [comment]
Great questions @studioswong!

1. Nothing too magic here I'm afraid - the node version is changed in the custom `.gitpod.Dockerfile` that I wrote. 
2. There's nothing more to do! Yetu already added the Gitpod app to kedro-org. This PR contains all the configuration (in the `prebuild` section of gitpod.yml) that tells Gitpod when to trigger the prebuild and will put the "Review in Gitpod" badge in PR descriptions. If it all seems to work smoothly then I'll update the contributors guide and that's it.
3. Nothing! 
 
Just to be clear, you don't even need this PR to merge in order to use Gitpod on this repo. You could already review this PR in Gitpod by going to `https://gitpod.io#github.com/kedro-org/kedro-viz/pull/735`. What this PR adds to that is:
* the ability to do prebuilds so that dependencies are installed in advanced of starting the workspace
* the custom configuration (like Node version, automatically installing the Python dependencies, running the backend server)

**Review by studioswong — APPROVED** [review]
great stuff - thanks for setting this up 💯

**Pull request merged**